model/schema/core.yaml

```diff
@@ -82,28 +82,65 @@ classes:
       - internal
       - obsolete
 
-  BiologicalEntity:
+  CurieAuditedObject:
     is_a: AuditedObject
     description: >-
-      An entity of biological origin that can be unambiguously attributed to
-      a single species.
-    abstract: true
+      Parent class for all LinkML classes with a curie
     slots:
       - curie
+
+  SubmittedCurieObject:
+    is_a: CurieAuditedObject
+    description: >-
+      Parent class for all LinkML classes of submitted entities or annotations
+      that have (or will have) a curie generated at the Alliance
+    slots:
+      - mod_entity_id
+      - mod_internal_id
+    rules:
+      - postconditions:
```

Comment:
@markquintontulloch 
We need to consider the fact that currently the DiseaseAnnotation class has fields for curie, mod_entity_id, and mod_internal_id but SGD, ZFIN, RGD, and OMIM (via RGD) do not provide any of those fields. We only have a generated internal UniqueID. This was a requirement of disease annotation submission discussed early on.

Reply:
Sure, will remove the rules to cover that case